client/src/Try/Container.purs

```diff
@@ -442,6 +438,36 @@ renderCompilerErrors errors = do
     , renderPlaintext message
     ]
 
+shareButton :: forall q i o. H.Component q i o Aff
+shareButton =  H.mkComponent
+  { initialState: \_ -> 0
+  , render
+  , eval: H.mkEval $ H.defaultEval
+      { handleAction = handleAction
+      }
+  }
+  where 
+  handleAction :: Unit -> H.HalogenM Int Unit () o Aff Unit
+  handleAction _ = do
+      url <- H.liftEffect $ window >>= location >>= href
+      H.liftAff $ makeAff \f -> do
+        runEffectFn3 copyToClipboard url (f (Right unit)) (f (Left $ Aff.error "Failed to copy to clipboard"))
```

Comment:
This is copying the pattern I saw with how `setupIFrame` is called, but maybe showing a message to the user is more appropriate than throwing an `Error` here.

Reply:
Perhaps a simple `window >>= alert` would be a cheap-and-easy way to notify the user? I know it's not the best UX, but do we anticipate this failing that often?

Reply:
Actually, since the "Share URL" changes to 'Copied to Clipboard" when that is clicked, could it also be changed to "Failed to Copy" when a failure occurs?

Reply:
> Actually, since the "Share URL" changes to 'Copied to Clipboard" when that is clicked, could it also be changed to "Failed to Copy" when a failure occurs?

Yeah, I'll plan on that 👍

Reply:
Showing the error message in UI here 6eb73be18a074c3f82065eabda4e450e981e77a7

Reply:
> do we anticipate this failing that often?

Browser support is pretty much universal https://caniuse.com/mdn-api_clipboard_writetext. I was thinking there was a possibility that some users might have disabled writing to the clipboard, but I haven't figured out how to do that in Chrome/Firefox. In Chrome there is a setting for disabling reading from clipboard, but haven't seen a way of limiting write access.

Reply:
Hmm, I’d probably want to also wrap the FFI call in `try`/`catch` in the 5% chance that `navigator.clipboard.writeText` doesn’t exist (in non-supported browsers)

Reply:
Updated here: [973b0ff](https://github.com/purescript/trypurescript/pull/299/commits/973b0ff508cc85e88eef7b0f770db35e3c53bebf)